From c11fe4763a7bfcf86ae33de5313c5c090d8cf4dc Mon Sep 17 00:00:00 2001
From: Kaifeng Zhu <kaifeng.zhu@citrix.com>
Date: Thu, 8 Jan 2015 06:14:37 +0000
Subject: [PATCH] CP-10662: Fix switch pciemulation issue

PCI emulation didn't work as it didn't parse the -pciemulation flags
before setting up the PCI devices (in function pc_init1 in hw/pc.c which
is called by machine->init in main).

Moving ahead the parsing code could solve the issue.
---
 vl.c |   18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/vl.c b/vl.c
index cccd85e..fa6bd40 100644
--- a/vl.c
+++ b/vl.c
@@ -5557,6 +5557,15 @@ int main(int argc, char **argv, char **envp)
         }
     }
 
+#ifdef CONFIG_PASSTHROUGH
+    for (i = 0; i < nb_pci_emulation; i++) {
+        if (pci_emulation_add(pci_emulation_config_text[i]) < 0) {
+            fprintf(stderr, "Warning: could not add PCI device %s\n",
+                    pci_emulation_config_text[i]);
+        }
+    }
+#endif
+
     machine->init(ram_size, vga_ram_size, boot_devices,
                   kernel_filename, kernel_cmdline, initrd_filename, cpu_model,
 		  direct_pci);
@@ -5683,15 +5692,6 @@ int main(int argc, char **argv, char **envp)
         }
     }
 
-#ifdef CONFIG_PASSTHROUGH
-    for (i = 0; i < nb_pci_emulation; i++) {
-        if (pci_emulation_add(pci_emulation_config_text[i]) < 0) {
-            fprintf(stderr, "Warning: could not add PCI device %s\n",
-                    pci_emulation_config_text[i]);
-        }
-    }
-#endif
-
     for(i = 0; i < MAX_VIRTIO_CONSOLES; i++) {
         const char *devname = virtio_consoles[i];
         if (virtcon_hds[i] && devname) {
-- 
1.7.10.4

